src/fifo/FifoControllerIndirect.cpp

```diff
@@ -24,6 +24,7 @@ FifoControllerIndirect::FifoControllerIndirect(uint32_t numFrames,
                                                int64_t * readCounterAddress,
                                                int64_t * writeCounterAddress)
         : FifoControllerBase(numFrames, threshold)
+        // Why are we casting `int64_t *` to `std::atomic<uint64_t> *`?
```

Comment:
This shouldn't work. However, nowhere in the code do we have a FifoBuffer that uses FifoControllerIndirect.

I'm going to update FifoBuffer/FifoControllberIndirect to take an atomic directly then.